File: webapi/Models/Citizen/Citizen.cs
using System.Collections.Generic;

namespace web_api.Models.Citizen
{
    public class Citizen : User
    {
        public string Name { get; set; }
        public string Address { get; set; }
        public string PhotoUrl { get; set; }
        public int Group { get; set; } 
        public bool IsSponsorRequired { get; set; }
        public bool SupportEfforOf { get; set; }
        public bool SponsorStudent { get; set; }
        public MemorizationList MemorizationList { get; set; }

        public Citizen()
        {
            MemorizationList = new MemorizationList();
        }
    }
}

File: angular_seed/src/app/core/citizen.ts
export class Citizen {
    id: number;
    pledge: number;
    name: string;
    email: string;
    password: string;
    username: string;
    group: number;
    photoUrl: string;
    profile_type: number;
    isLookingForSponsor: boolean;
    supportEfforOf: boolean;
    sponsorStudent: boolean;
    address: string;
    guId: string;


    constructor() {
        this.pledge = -1;
        this.name = '';
        this.email = '';
        this.username = '';
        this.group = -1;
        this.photoUrl = '';
        this.profile_type = -1;
        this.address = '';
        this.isLookingForSponsor = false;
        this.supportEfforOf = false;
        this.sponsorStudent = false;
    }

    validate() {
        if (this.pledge === -1 || this.name === ''
            || this.email === '' || this.username === ''
            || this.password === '' || this.group === -1 || this.profile_type === -1) {
            return false;
        }
        return true;
    }
}
